src/jit/gentree.cpp

```diff
@@ -394,6 +408,88 @@ bool GenTree::IsNodeProperlySized() const
 }
 #endif
 
+/*****************************************************************************
+ *
+ *  When 'NODEBASH_STATS' is enabled in "jit.h" we record all instances of
+ *  an existing GenTree node having its operator changed. This can be useful
+ *  for two (related) things - to see what is being bashed (and what isn't),
+ *  and to verify that the existing choices for what nodes are marked 'large'
+ *  are reasonable (to minimize "wasted" space).
+ *
+ *  And yes, the hash function / logic is simplistic, but it is conflict-free
+ *  and transparent for what we need.
+ */
+
+#if NODEBASH_STATS
+
+#define BASH_HASH_SIZE 211
+
+inline hashme(genTreeOps op1, genTreeOps op2) { return ((op1 * 104729) ^ (op2 * 56569)) % BASH_HASH_SIZE; }
+
+struct BashHashDsc
+{
+    unsigned __int32    bhFullHash; // the hash value (unique for all old->new pairs)
+    unsigned __int32    bhCount;    // the same old->new bashings seen so far
+    unsigned __int8     bhOperOld;  // original gtOper
+    unsigned __int8     bhOperNew;  // new      gtOper
+};
+
+static  BashHashDsc BashHash[BASH_HASH_SIZE];
+
+void                GenTree::RecordOperBashing(genTreeOps operOld, genTreeOps operNew)
+{
+    unsigned        hash = hashme(operOld, operNew);
+    BashHashDsc    *desc = BashHash + hash;
+
+    if (desc->bhFullHash != hash)
+    {
+        noway_assert(desc->bhCount == 0);   // if this ever fires, need to fix the hash fn
+        desc->bhFullHash = hash;
+    }
+
+    desc->bhCount   += 1;
+    desc->bhOperOld  = operOld;
+    desc->bhOperNew  = operNew;
+}
+
+void                GenTree::ReportOperBashing(FILE *f)
+{
+    unsigned        total = 0;
+
+    fflush(f);
```

Comment:
I'm glad you asked - there are several components that spew stuff to stdout and the output sometimes gets intermingled. It's always just debug spew (i.e. a "real" user would never see any of it) and it only happens when you build with several off-by-default / apparently never used #ifdef's enabled, and so one could argue that things are "good enough" as is (with the flush the mixing happens a lot less and, frankly, I didn't dig deeper into it) but if folks think this is worth investigating further let's definitely do so.